### PR TITLE
MM-9963: test deactivation

### PR DESCRIPTION
**Pull request by lieut-data**
This started as an extension of the tests to consider the case of a deactivated user. This in turn prompted the ability to re-activate a user, and then to coordinate among other running load test instances.

The biggest change here is the creation and use of a LoadtestInstances table in the same database under test, in which each starting instance registers itself and selects the appropriate subset of users to which to login with entities. The logic around what user types to start is also simplified to use our `randutil.Choice`, making it a random selection based on the configured distribution. The only outstanding item is to allow randomization of users across load test instances (by converging on a uniform random seed) -- right now, there would be overlap, but I'm not sure this feature is highly used at present.

I also made `loadtestconfig.json` an ignored file, with a `loadtestconfig.default.json` to take its place.

Lastly, I switched over to sqlx to simplify abstracting around MySQL and PostgreSQL.

Credit to @jwilander for some overlapping code in https://github.com/mattermost/mattermost-load-test/compare/deactivating-users.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,13 +7,16 @@ DIST_ROOT=dist
 DIST_FOLDER_NAME=mattermost-load-test
 DIST_PATH=$(DIST_ROOT)/$(DIST_FOLDER_NAME)
 
-
 all: install
 
 install:
+	dep ensure
```
Comment (lieut-data) [inline Makefile:13]: The documentation suggested modifying the glide (now dep) files to point at the desired branch, but nothing triggered a reevaluation of the dependencies.

File: loadtest/client.go [hunk 2]
```diff
@@ -21,26 +21,31 @@ func newClientFromToken(token string, serverUrl string) *model.Client4 {
...
+	ThreadSplit(cfg.UserEntitiesConfiguration.NumActiveEntities, runtime.GOMAXPROCS(0)*2, PrintCounter, func(i int) {
 		// Add the usernum to start from
-		userNum := entityNum + cfg.UserEntitiesConfiguration.EntityStartNum
+		entityNum := i + entityStartNum
+		userNum := entityNum
 		client := model.NewAPIv4Client(cfg.ConnectionConfiguration.ServerURL)
 
-		// Random selection if picked
+		// Random selection if picked.
+		// TODO: Synchronize random selection across multiple loadtest instances by using
```
Comment (lieut-data) [inline loadtest/client.go:36]: I'm going to circle back on this, but wanted to get the meat of the changes up for review.

File: loadtest/config.go [hunk 3]
```diff
@@ -26,7 +26,6 @@ type LoadTestConfig struct {
 type UserEntitiesConfiguration struct {
 	TestLengthMinutes                 int
 	NumActiveEntities                 int
-	EntityStartNum                    int
```
Comment (lieut-data) [inline loadtest/config.go:29]: This is now determined automatically.

File: loadtest/run.go [hunk 4]
```diff
@@ -20,10 +19,14 @@ import (
 
 	"github.com/gizak/termui"
 	"github.com/mattermost/mattermost-load-test/cmdlog"
+	"github.com/mattermost/mattermost-load-test/randutil"
 	"github.com/mattermost/mattermost-server/model"
 )
 
 func RunTest(test *TestRun) error {
+	interruptChannel := make(chan os.Signal)
```
Comment (lieut-data) [inline loadtest/run.go:27]: Moved this up so I can handle interrupts while still starting entities.

File: loadtest/run.go [hunk 5]
```diff
@@ -82,82 +114,89 @@ func RunTest(test *TestRun) error {
...
 	numEntities := len(tokens)
-	entityNum := 0
-	entitiesToSkip := cfg.UserEntitiesConfiguration.EntityStartNum
-	for _, usertype := range test.UserEntities {
-		numEntitesToCreateForType := int(math.Floor((float64(usertype.Freq) / 100.0) * float64(numEntities)))
-		startEntity := 0
-		if numEntitesToCreateForType <= entitiesToSkip {
-			entitiesToSkip -= numEntitesToCreateForType
+	cmdlog.Infof("Starting %d entities from offset %d", numEntities, loadtestInstance.EntityStartNum)
+	for i := 0; i < numEntities; i++ {
```
Comment (lieut-data) [inline loadtest/run.go:128]: I found the previous strategy of iterating over the UserEntities and counting how many types to create difficult to adapt to this new pattern of coordinating among multiple load test instances. Now, just pick one at random according to the configured distribution.

File: loadtest/client.go (hunk 2 above)
Comment (crspeller) [reply loadtest/client.go:36]: I don't think this works now without screwing things up :P

**Review by crspeller — APPROVED** [review]

**Pull request merged**